src/platform/storage/zephyr/nvs/storage.c

```diff
@@ -384,10 +384,24 @@ mender_storage_get_artifact_name(const char **artifact_name) {
     mender_err_t ret = nvs_read_alloc(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_ARTICACT_NAME, (void **)artifact_name, &artifact_name_length);
     if (MENDER_OK != ret) {
         if (MENDER_NOT_FOUND == ret) {
-            if (NULL == (*artifact_name = mender_utils_strdup("unknown"))) {
-                mender_log_error("Unable to allocate memory");
-                return MENDER_FAIL;
+
+            /* Get the Artifact Name from the build, if set */
+#ifdef CONFIG_MENDER_ARTIFACT_NAME
+            if (0 < strlen(CONFIG_MENDER_ARTIFACT_NAME)) {
+                if (NULL == (*artifact_name = mender_utils_strdup(CONFIG_MENDER_ARTIFACT_NAME))) {
+                    mender_log_error("Unable to allocate memory");
+                    return MENDER_FAIL;
+                }
+            } else {
+#endif
+                if (NULL == (*artifact_name = mender_utils_strdup("unknown"))) {
+                    mender_log_error("Unable to allocate memory");
+                    return MENDER_FAIL;
+                }
```

Comment:
You could first assign the right literal to a `const char *artf_name_literal` and then do `mender_utils_strdup(artf_name_literal)`. Could play nicer with the `#ifdef...#endif`. But YMMV.

Reply:
I tried that, but my C skills failed me. I'll try again...

In any case an ifdef will stay because the macro is not always defined.

Reply:
Success.